### PR TITLE
CTCP-5488 Unloading new procedure submission logic

### PR DESCRIPTION
**Pull request by Reece-Carruthers**

### CODE REVIEW

**Comment by Reece-Carruthers** [comment]
[screen-recorder-wed-jul-24-2024-08-14-12.webm](https://github.com/user-attachments/assets/efe830c1-92db-46f0-a58d-a433b82b4ceb)

**Comment by Reece-Carruthers** [comment]
image

File: app/services/submission/SubmissionService.scala [hunk 1]
```diff
@@ -107,22 +107,37 @@ class SubmissionService @Inject() (
...
-    } yield UnloadingRemarkType(
-      conform = conform,
-      unloadingCompletion = unloadingCompletion,
-      unloadingDate = unloadingDate,
-      stateOfSeals = stateOfSeals,
-      unloadingRemark = unloadingRemark
-    )
+    def generateUnloadingRemarkForRevisedProcedureYes: Reads[UnloadingRemarkType] =
+      UnloadingRemarkType(
+        conform = Number1,
```
[the remark this answers is not in the document]
Comment (jackheslop96) [reply app/services/submission/SubmissionService.scala:112]: Not a bad suggestion. Could just add an alias `import generated.{Number1 => True, _}` for the first two.

The seals stuff we could make use of the StateOfSeals case class. Maybe just add `val PresentAndNotDamaged: StateOfSeals = StateOfSeals(Some(true))` to the companion object

[the remark this answers is not in the document]
Comment (Reece-Carruthers) [reply app/services/submission/SubmissionService.scala]: @jackheslop96 that might be a better way of doing it than the one in the latest commit, as what I've done seems a little overkill for what it is. What do you think?

**Pull request merged**